src/main/java/app/bpartners/geojobs/endpoint/rest/controller/mapper/DetectableObjectTypeMapper.java

```diff
@@ -74,33 +75,55 @@ public DetectableObjectType toRest(DetectableType domain) {
 
   public List<DetectableObjectType> mapFromModel(Object o) {
     var objectTypes = new ArrayList<DetectableObjectType>();
-    if (o instanceof BPToitureModel) {
-      BPToitureModel model = (BPToitureModel) o;
-      if (model.getArbre().booleanValue()) {
-        objectTypes.add(ARBRE);
-      }
-      if (model.getToitureRevetement().booleanValue()) {
-        objectTypes.add(TOITURE_REVETEMENT);
-      }
-      if (model.getPanneauPhotovoltaique().booleanValue()) {
-        objectTypes.add(PANNEAU_PHOTOVOLTAIQUE);
-      }
-      // TODO: add other detectableTypes in BPToitureModel not handled yet
-    } else if (o instanceof BPLomModel) {
-      BPLomModel model = (BPLomModel) o;
-      if (model.getPassagePieton().booleanValue()) {
-        objectTypes.add(PASSAGE_PIETON);
-      }
-      if (model.getTrottoir().booleanValue()) {
-        objectTypes.add(TROTTOIR);
-      }
-      // TODO: add other detectableTypes in BPLomModel not handled yet
+
+    if (o instanceof BPToitureModel model) {
+      mapFromBPToitureModel(model, objectTypes);
+    } else if (o instanceof BPLomModel model) {
+      mapFromBPLomModel(model, objectTypes);
+    } else if (o instanceof BPZanModel model) {
+      mapFromBPZanModel(model, objectTypes);
     } else {
       throw new ApiException(SERVER_EXCEPTION, "Unknown instance of object " + o.getClass());
     }
+
     return objectTypes;
   }
 
+  private void mapFromBPToitureModel(BPToitureModel model, List<DetectableObjectType> objectTypes) {
```

Comment:
No. 

As I said, return List<DetectableObjectType> here and add the returned list to the main List inside mapFromModel function.